src/ui/buttons/props.js

```diff
@@ -694,13 +697,43 @@ export function normalizeButtonStyle(
       BUTTON_SIZE_STYLE[BUTTON_SIZE.HUGE].maxHeight,
     ];
 
+    if (disableMaxHeight === true) {
+      throw new TypeError(
+        `Unexpected style.height for style.disableMaxHeight: got: ${height}, expected undefined.`
+      );
+    }
+
     if (height < minHeight || height > maxHeight) {
       throw new Error(
         `Expected style.height to be between ${minHeight}px and ${maxHeight}px - got ${height}px`
       );
     }
   }
 
+  if (disableMaxHeight !== undefined) {
+    if (typeof disableMaxHeight !== "boolean") {
+      throw new TypeError(
+        `Expected style.disableMaxHeight to be a boolean, got: ${disableMaxHeight}`
+      );
+    }
+
+    const disableMaxHeightInvalidFundingSources = [FUNDING.CARD, undefined];
+    const disableMaxHeightValidFundingSources = [
```

Comment:
why are these the only ones supported for disabling max height?

Reply:
The only 2 buttons that aren't supported are the smart stack and card. The main reason these aren't supported was the challenge of getting them to render correctly and not being the specific use case that Stripe was looking for.

It seemed like they were rendering the standalone paypal button so we focused on enabling that only.

The card button is not supported because the inline guest experience gets rendered in the same container as the card button. If the merchant but a height limitation on the container for the card button, then it was also going to apply to the inline guest experience.

I know we have a bunch of other funding sources, but I believe we only have buttons for the funding sources below. The other funding sources are for the card fields, correct?

![Screenshot 2024-05-21 at 9 24 12 AM](https://github.com/paypal/paypal-checkout-components/assets/32575209/63596750-dd43-4235-976a-b2cb75a40b3a)

Reply:
We also have all the APMs although maybe we don't support them as standalone buttons? If so, I would think we would have a config value somewhere that disallows those buttons as standalone

https://github.com/paypal/paypal-sdk-constants/blob/main/src/funding.js#L3

Reply:
Looks like all those funding sources can be rendered as a standalone button. I've updated the error message to include all of those as well.

Reply:
I would consider the following tweaks, might be easier to maintain:

1. Early return if `fundingSource === undefined`. Error message could be something like `disableMaxHeight is only supported for standalone buttons. You can render a standalone button by passing `fundingSource` to the `Button` config: paypal.Buttons({fundingSource: paypal.FUNDING.PAYPAL})

2. Then, down below, throw the error for invalid buttons and just say "This funding source, {fundingSource}, does not support `disableMaxHeight`"

That's some very rough copy, feel free to change it if you decide to go in this direction